### DISCUSSION TITLE
How to get the value you're mapping from

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#769] [General]
I have a mapping s_tokenToPriceFeed from address token to address price feed. I know I can use it to get the price feed (s_tokenToPriceFeed(token)) but can I also use the same mapping to get token if I input price feed?

Or do I have to create an array of tokens and get it that way?

**Comment by theirrationalone** [comment]
@0xwtfwtf !, I'll recommend you to use struct with array.

```solidity
contract PlayStruct {
    struct TokensAndPriceFeeds {
        address token;
        address priceFeed;
    }

    TokensAndPriceFeeds public tokenPriceFeed;
    TokenAndPriceFeeds[] public tokensPriceFeedsList;

    function storeData(address _token, address _priceFeed) public {
        tokenPriceFeed = TokensAndPriceFeeds(__token, _priceFeed);
    }

    function storeDataIntoList(address _token, address _priceFeed) public {
        TokensAndPriceFeeds memory dataGen = TokensAndPriceFeeds(__token, _priceFeed);
        tokensPriceFeedsList.push(dataGen);
    }
}
```

Below is a pseudo code i don't claim it may right, there can have some syntactic errors. 😉
